New documentation content (#671)
graknlabs/docs: 0001-typedb/02-dev/02-schema.md

---
pageTitle: Schema
keywords: typeql, schema, type hierarchy, reserved keywords
longTailKeywords: typeql schema, typeql type hierarchy, typeql data model, typeql reserved keyword
Summary: Introduction to the TypeDB Schema.
---

# Schema

TypeDB database stores data and schema. 

> TypeDB schema is a blueprint of a database. It describes how the data can be structured: 

Reviewer: ```suggestion
A TypeDB schema is the blueprint of a database. It describes how the data can be structured: 
```

- All user-defined types. 
- Type hierarchy. 
- Data constraints. 
- Inference rules. 

With TypeDB, thanks to our high-level knowledge model, the schema closely resembles 
[conceptual schema](https://en.wikipedia.org/wiki/Conceptual_schema). TypeDB takes care of the hassle of going through

> other modelling steps. This greatly simplifies the design process, providing us with, what can be considered, a highly 

Reviewer: ```suggestion
other modeling steps. This greatly simplifies the design process, providing us with a highly 
```

> normalised distributed schema without the need of going through logical and physical modelling.

Reviewer: ```suggestion
normalized distributed schema without the need of going through logical and physical modelling.
```
